Makefile

```diff
@@ -1044,7 +1040,7 @@ grpc-teleterm:
 # Unlike grpc-teleterm, this target runs locally.
 .PHONY: grpc-teleterm/host
 grpc-teleterm/host: protos/all
-	cd lib/teleterm && $(BUF) generate
+	$(BUF) generate --template=lib/teleterm/buf.gen.yaml lib/teleterm/api/proto
```

Comment:
I'm not sure but I think I should move those `buf.gen.yaml` files into the directories defined in `buf.work.yaml`.

`buf.gen.yaml` files used to be defined on the same level as the old `buf.work.yaml`. Since we removed those extra workspaces, it'd probably make more sence to have `buf.gen.yaml` on the same level as `buf.yaml`. https://docs.buf.build/configuration/v1/buf-gen-yaml

Reply:
My 2c is to keep buf.gens besides the workspace file, as it is the same path were we expect codegen commands to start from.

I suggest doing away with all the additional buf.gen files and using go_package to control the output location, like we do for for everything under proto/ today. We can control whether to use protoc-gen-go or gogo in genproto.sh. (FYI @timothyb89 for prehog.)